bundles/org.openhab.core.model.thing/src/org/openhab/core/model/thing/internal/GenericItemChannelLinkProvider.java

```diff
@@ -98,15 +99,16 @@ private void createItemChannelLink(String context, String itemName, String chann
             previousItemNames.remove(itemName);
         }
 
-        Set<ItemChannelLink> links = itemChannelLinkMap.get(itemName);
+        Map<ChannelUID, ItemChannelLink> links = itemChannelLinkMap.get(itemName);
         if (links == null) {
-            itemChannelLinkMap.put(itemName, links = new HashSet<>());
+            itemChannelLinkMap.put(itemName, links = new HashMap<>(2));
```

Comment:
```suggestion
            itemChannelLinkMap.put(itemName, links = new HashMap<>(2));
```

Why `2`?

Reply:
The default is 16 with .75 load factor. That seems like a waste of memory because most items have just one link.

If I set it to one, as soon as one element is added, it would trigger a rehash and capacity expansion anyway, because it would have exceeded the load factor. So 2 is chosen here.

Tbh this is just a guess based on reading about the description of initial capacity and load factor from hashmap's doc. I haven't done actual memory analysis tests to verify.

Reply:
I've added a comment in the code for future reference